File: src/007/main.c
# include <stdbool.h>
# include <stdio.h>

typedef unsigned int u32;

bool is_prime (u32 number) {
	for (u32 i = 2; i * i <= number; i++) {
		if (number % i == 0) {
			return false;
		}
	}
	return true;
}

int main () {
	const u32 nth = 10001;
	u32 number = 1;
	
	for (u32 i = 1; i < nth; ) {
		number += 2;
		if (is_prime (number)) {
			i++;
		}
	}
	
	printf ("%u\n", number);
}

File: src/007/main.rs

fn is_prime (number: u32) -> bool {
	let mut i: u32 = 2;
	while i * i <= number {
		if number % i == 0 {
			return false;
		}
		i += 1;
	}
	return true;
}

fn main () {
	const NTH: u32 = 10001;
	let mut number: u32 = 1;
	let mut i: u32 = 1;
	
	while i < NTH {
		number += 2;
		if is_prime (number) {
			i += 1;
		}
	}
	
	println! ("{}", number);
}

